From 0c86ab1ba7c469ddf1b2738c07ca30a5f66c670d Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 29 Nov 2017 13:34:19 +0200
Subject: [PATCH 2937/5242] MLK-17032-1: arm64: dts: imx8mq: Fix DCSS
 suspend/resume issue

commit  04a034a3fc5e5f6cf294866ab2b8c0b8732c943b from
https://source.codeaurora.org/external/imx/linux-imx.git

This commit:

6362b8c - MLK-17014-2 ARM64: dts: imx8mq: move 'display-subsystem' node
          to dtsi file

moved the display-subsystem node to dtsi, before the DCSS node definition. This
made the DCSS core suspend/resume after display-subsystem. However, the DCSS
clocks need to be enabled first.

This patch moves the display-subsytem definition after the DCSS and HDMI PHY nodes.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 026d363..fdb0b67 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -446,11 +446,6 @@
 		clock-names = "ipg", "per", "osc_per";
 	};
 
-	display-subsystem {
-		compatible = "fsl,imx-display-subsystem";
-		ports = <&dcss_disp0>;
-	};
-
 	irqsteer_dcss: irqsteer@32e2d000 {
 		compatible = "nxp,imx-irqsteer";
 		reg = <0x0 0x32e2d000 0x0 0x1000>;
@@ -602,6 +597,11 @@
 		status = "disabled";
 	};
 
+	display-subsystem {
+		compatible = "fsl,imx-display-subsystem";
+		ports = <&dcss_disp0>;
+	};
+
 	mipi_dsi_phy: dsi_phy@30A00300 {
 		#address-cells = <1>;
 		#size-cells = <0>;
-- 
1.7.9.5

